tests/Type/data/eloquent-builder.php

```diff
@@ -342,6 +343,13 @@ function test(
     assertType('Illuminate\Database\Eloquent\Builder<App\Team|App\User>', $userOrTeamBuilder->where('id', 5));
 
     assertType('Illuminate\Database\Eloquent\Builder<TModel of Illuminate\Database\Eloquent\Model (function EloquentBuilder\test(), argument)>', $templateBuilder->select());
+
+    if (laravel_version_compare('12.15.0', '<')) {
+        assertType('Illuminate\Database\Query\Builder', User::query()->getCountForPagination());
+    }
+    else {
+        assertType('int', User::query()->getCountForPagination());
+    }
```

Comment:
As a convention, we do not do the condition check inside the test cases like this.As it wouldn't work correctly.  We do this inside the data provider. In this case in `GeneralTypeTest` There you can also see how it is done for example.

Reply:
> How do I correctly resolve the application version? Is it safe to use this inside of the BuilderHelper@__construct() method?

Aah, I was referring to source code with the `laravel_version_compare` suggestion. Sorry for confusion. Inside the `src` directory we can use `laravel_version_compare` function. I think we already do it in one place, you can search for it.

> GeneralTypeTest: should I create a new eloquent-builder-l12-15.php for this type assertion?
Yes, please. And conditionally `yield` depending on version.